### PR TITLE
Remove snake oil reference with Sos App

### LINKED ISSUE #157: Remove all references to "Snake oil software"
As per the initial feedback we are getting, people don't appreciate the humor :-(

### PR DESCRIPTION
**Pull request by jkhabra**
Resolves #157
Fix image upload issue ( user can submit the form even before image is uploaded )

### CODE REVIEW

File: ui/src/components/Employee/Create/index.tsx [hunk 1]
```diff
@@ -31,6 +32,10 @@ const CreateEmployeeForm: React.FC<FormikProps<CreateEmployeeFormValues> & Creat
   p,
 ) => {
   const [isModalOpen, setModalOpen] = useState<boolean>(false);
+  const [uploadStatus, setUploadStatus] = useState<boolean>(false);
+  const hadleUploadStatus = (status: boolean) => {
+    setUploadStatus(status);
+  };
```
Comment (bitspook) [inline ui/src/components/Employee/Create/index.tsx:39]: These don't look like they are related to "Remove snake oil software" references. Occasional exception is alright, but please don't make it a norm to mix arbitrary changes in unrelated PRs. One branch per task, and one PR per branch. Mixing changes like this leads up to very bad development practices very fast.

File: ui/src/components/Employee/Create/index.tsx [hunk 2]
```diff
@@ -89,7 +94,12 @@ const CreateEmployeeForm: React.FC<FormikProps<CreateEmployeeFormValues> & Creat
 
             <div className={c('logo', 'field-container')}>
               <span className={c('field-title')}>Head Shot</span>
-              <ImageUploadField className={c('image-container')} type={'file'} name="headshot" />
+              <ImageUploadField
+                className={c('image-container')}
+                type={'file'}
+                name="headshot"
+                uploadStatus={hadleUploadStatus}
```
Comment (bitspook) [inline ui/src/components/Employee/Create/index.tsx]: Why do we need to maintain a separate upload status? Can't we validate on absense of `form.values.headshot`?

File: ui/src/components/Form/ImageUploadField/index.tsx [hunk 3]
```diff
@@ -16,6 +16,7 @@ interface Props {
   name: string;
   className?: string;
   type?: string;
+  uploadStatus: any;
```
Comment (bitspook) [inline ui/src/components/Form/ImageUploadField/index.tsx]: Please stop using `any` for types.

File: ui/src/pages/dashboard/index.tsx [hunk 4]
```diff
@@ -50,7 +51,7 @@ const Dashboard = () => {
 
   return (
     <DashboardLayout
-      title={'Dashboard - Snake Oil Software'}
+      title={`Dashboard - ${config.appName}`}
```
Comment (bitspook) [inline ui/src/pages/dashboard/index.tsx:54]: Good thinking with getting the name from config :+1:

**Review by bitspook — CHANGES_REQUESTED** [review]
More than half of the changes are related to the upload status bug. Please don't mix unrelated issues in a single branch moving forward.

File: ui/src/components/Employee/Create/index.tsx (hunk 2 above)
Comment (jkhabra) [reply ui/src/components/Employee/Create/index.tsx]: Image-upload-field is optional field so that's why we need to maintain separate upload status.

**Review by bitspook — APPROVED** [review]

**Pull request merged**